File: rh-front/src/app/models/mini_message.ts
import { User } from "./user";

export class MiniMessage {
    hasUnseenMessage?: boolean;
    lastMessageText?: string;
    date?: Date;

    otherUser?: User;
    // constructor(h: boolean | undefined, l: string | undefined, d: Date | undefined, o: User | undefined) {
    //     this.hasUnseenMessage = h;
    //     this.lastMessageText = l;
    //     this.date = d;
    //     this.otherUser = o;
    // }
}

File: messagerie/src/main/java/com/rh/messagerie/entities/MiniMessage.java
package com.rh.messagerie.entities;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.Date;

@Data
@AllArgsConstructor
public class MiniMessage {
    boolean hasUnseenMessage;
    String lastMessageText;
    Date date;

    User otherUser;
}
